plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidator.java

```diff
@@ -144,6 +149,53 @@ private void validateServletMapping() {
     }
   }
   
+  /**
+   * Verifies that every <jsp-file> element exists in the project.
+   */
+  private void validateJsp() {
+    IProject project = resource.getProject();
```

Comment:
You should first check if the servlet-version is 2.5. If it isn't, just skip the rest of this.